### DISCUSSION TITLE
Efficient term reweighting

### DISCUSSION THREAD

**Discussion by chenyn66** [castorini/pyserini#1269] [Q&A]
Hi,
I'm trying to do term reweighting with the Query Builder API. However, it looks like using this API makes the searching much slower. Is this expected? If so, is there any efficient way to reweight terms in a query?
Thanks

**Comment by lintool** [comment]
Efficient query evaluation depends on being able to skip "work" computing scores that can never make it into the top-_k_. This is dependent on the term scores and their distribution. It may be the case that weighting terms changes the term score distributions such that these optimizations are no longer possible.

Can you provide some examples of queries and query latencies?

**Reply by chenyn66** [reply]
Thanks for the reply! But I think I have figured it out. The main issue is the increasing query length of my reweighting algorithm.
